### PR TITLE
feat: logout and refreshtoken, refactor

### PR DESCRIPTION
**Pull request by yurcco**

### CODE REVIEW

File: Sloth.Api.sln [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30011.22
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Sloth.Api", "src\Sloth.Api\Sloth.Api.csproj", "{7C86C9DF-EA83-4781-9AB8-73D18984E9DC}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "db", "db", "{271B69BB-2D9B-4D02-8BE8-965A9435B018}"
 	ProjectSection(SolutionItems) = preProject
 		db\V1.0__create_tables.sql = db\V1.0__create_tables.sql
 		db\V1.1__add_chat_columns.sql = db\V1.1__add_chat_columns.sql
+		V1.2_add_user_email_colum_and_sessionrefreshtoken_table.sql = V1.2_add_user_email_colum_and_sessionrefreshtoken_table.sql
```
Comment (yurcco) [inline Sloth.Api.sln]: Db

File: V1.2_add_user_email_colum_and_sessionrefreshtoken_table.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+ALTER TABLE [dbo].[User] ADD [Email] VARCHAR (50) NULL;
+GO
+
+CREATE TABLE [dbo].[SessionRefreshToken](
+	[Id] [uniqueidentifier] NOT NULL PRIMARY KEY DEFAULT NEWID(),
+	[UserId] [uniqueidentifier] NOT NULL,	
+	[Token] [nvarchar](500) NULL,
+	[ExpiredTime] [datetime] NULL,
+	[IsActive] [bit] NULL,
```
Comment (yurcco) [inline V1.2_add_user_email_colum_and_sessionrefreshtoken_table.sql]: Not null

File: src/Sloth.Api/Controllers/UserController.cs [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
         }
 
-        /// <summary>
-        /// Get all users  
-        /// </summary>
-        [HttpGet]
-        public async Task<User[]> Register()
+        [HttpGet("current")]
+        public async Task<CurrentUser> Current()
         {
```
Comment (yurcco) [inline src/Sloth.Api/Controllers/UserController.cs:24]: GetCurrentAsync

File: src/Sloth.Api/Services/IUserService.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using Sloth.Auth.Models;
+using System;
+using System.Threading.Tasks;
+
+namespace Sloth.Api.Services
+{
+    public interface IUserService
+    {
+        Task<CurrentUser> GetCurrentUserAsync(Guid guid);
```
Comment (yurcco) [inline src/Sloth.Api/Services/IUserService.cs]: UserId

File: src/Sloth.Auth/Bootstrap.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                    options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
+                    options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
+
+                })
+
+                .AddJwtBearer(cfg =>
+                {
+                    cfg.RequireHttpsMetadata = false;
+                    cfg.SaveToken = true;
+                    cfg.TokenValidationParameters = new TokenValidationParameters
```
Comment (yurcco) [inline src/Sloth.Auth/Bootstrap.cs]: Move to options

**Review by MTrofimenko — APPROVED** [review]

**Pull request merged**